server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -747,6 +758,10 @@ public HttpServer build() {
         }
         serverConfig = buildSpnegoConfiguration(this);
         break;
+      case CUSTOM:
+        serverConfig = buildCustomConfiguration(this);
```

Comment:
This indirection seems unnecessary. Let's not add this unless there's something this method is supporting :)

Reply:
This method helps in getting the `AvaticaServerConfiguration` object from the builder. Why do you not want to have it?

Reply:
Are you sure? I'm not sure this is doing anything. If I'm reading this right, this line is effectively:

```java
serverConfig = this.serverConfig;
```

You already did this in `withCustomAuthentication`, so I'm not sure what the intent is to do here. Please tell me if I'm missing something (I'm admittedly not looking at this in an editor, so it might just be me)

Reply:
Nice catch @joshelser , That's correct, its effectively the same line when I had a closer look on it.
I think I just created that method to keep it in sync with other Auth methods above. 
What do you want me to do? Remove the case from the switch?

Reply:
I would leave the switch case, but replace what you have presently with a comment telling future-us that we don't need to do any configuration building for `CUSTOM`.

Reply:
Okay, Just to clarify, I will keep the same code in place and add the comments.